byron/chain/executable-spec/src/Cardano/Spec/Chain/STS/Rule/Chain.hs

```diff
@@ -105,6 +152,95 @@ instance Embed BBODY CHAIN where
 instance Embed PBFT CHAIN where
   wrapFailed = PBFTFailure
 
+instance Embed (UTXOWS TxId) CHAIN where
+  wrapFailed = LedgerUTXOFailure
+
+instance Embed DELEG CHAIN where
+  wrapFailed = LedgerDelegationFailure
+
 genesisHash :: Hash
 -- Not sure we need a concrete hash in the specs ...
 genesisHash = Crypto.Hash.hash ("" :: ByteString)
+
+-- | Lens for the delegation interface state contained in the chain state.
+disL :: Lens' (State CHAIN) DIState
+disL = _5
+
+instance HasTrace CHAIN where
+
+  initEnvGen = (,,,) <$> gCurrentSlot <*> pure initialUTxO <*> initEnvGen @DELEG <*> gPParams
+    where
+      -- If we want to generate large traces, we need to set up the value of the
+      -- current slot to a sufficiently large value.
+      gCurrentSlot = Slot <$> Gen.integral (Range.constant 32768 2147483648)
+
+      -- TODO: For now we generate an empty UTxO. In later iterations we will
+      -- incorporate the UTxO payload into the chain.
+      initialUTxO = UTxO Map.empty
+
+      gPParams
+        = PParams
+        <$> gMaxHeaderSize
+        <*> gMaxBlockSize
+        <*> gMaxTxSize
+        <*> gMaxProposalSize
+        <*> gBlockSigCntThreshold
+        <*> gSlotsPerEpoch
+        <*> gUpdateProposalTTL
+        <*> pure 1 -- _scriptVersion
+        <*> gConfirmationTreshold
+        <*> gAdoptinThreshold
+        <*> pure k -- _stableAfter
+        <*> pure 0 -- factor a
+        <*> pure 0 -- factor b
+        where
+          -- In mainet the maximum header size is set to 2000000 and the maximum
+          -- block size is also set to 2000000, so we have to make sure we cover
+          -- those values here. The upper bound is arbitrary though.
+          gMaxHeaderSize = Gen.integral (Range.constant 0 4000000)
```

Comment:
For the lower bounds we should probably figure out the minimum block size. That should really be the size of a block containing only an update proposal to increase the size!

Reply:
Seems like a good idea. I added this to the comments as a `TODO`. I hope that's fine for now.